pkg/querier/remote_read.go

```diff
@@ -138,25 +145,36 @@ func remoteReadStreamedXORChunks(
 		http.Error(w, "internal http.ResponseWriter does not implement http.Flusher interface", http.StatusBadRequest)
 		return
 	}
-
-	w.Header().Set("Content-Type", "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse")
+	w.Header().Set("Content-Type", api.ContentTypeRemoteReadStreamedChunks)
 
 	for i, qr := range req.Queries {
 		if err := processReadStreamedQueryRequest(ctx, i, qr, q, w, f, maxBytesInFrame); err != nil {
-			if errors.Is(err, context.Canceled) {
-				// We're intentionally not logging in this case to reduce noise about an unactionable condition.
-				http.Error(w, err.Error(), statusClientClosedRequest)
-				return
+			code := remoteReadErrorStatusCode(err)
+			if code/100 != 4 {
+				level.Error(logger).Log("msg", "error while processing remote read request", "err", err)
 			}
-
-			level.Error(logger).Log("msg", "error while processing remote read request", "err", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			http.Error(w, err.Error(), code) // change the Content-Type to text/plain and return a human-readable error message
```

Comment:
[nit] I would remove the comment because it's not entirely true. It depends whether we've already sent the response headers or not. The error could happen at any time, even while sending chunks.